### PR TITLE
olexandr.vakulinskiy

### PR DESCRIPTION
**Pull request by AlexanderVak**

### CODE REVIEW

File: olexandr.vakulinskiy/PR1/practice.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+require 'faraday'
+
+class Image
+  def download(url, file_name)
+    response = Faraday.get url
+
+    raise ArgumentError, "wrong url, status code #{response.status}" unless response.status == 200
+    raise TypeError, 'file is not an image' unless image?(url[/\.\w{3,4}($|\?)/])
+
+    File.open("#{file_name}#{url[/\.\w{3,4}($|\?)/]}", 'wb') { |fp| fp.write(response.body) }
```
Comment (Flippylolz) [inline olexandr.vakulinskiy/PR1/practice.rb]: Extension will not be valid if provided url contains params like
https://miro.medium.com/max/4000/1*KUy_KKExZrSpBuv9XfyBgA.png?somasda=asd&location=Cherkassy

File: olexandr.vakulinskiy/PR1/practice.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  end
+
+  def image?(url_ext)
+    url_ext.eql?('.png') || url_ext.eql?('.jpg') || url_ext.eql?('.image') || url_ext.eql?('.jpeg')
```
Comment (Flippylolz) [inline olexandr.vakulinskiy/PR1/practice.rb]: nitpick: .image is not a valid extension

**Review by Flippylolz — CHANGES_REQUESTED** [review]

File: olexandr.vakulinskiy/PR1/practice.rb (hunk 1 above)
Comment (AlexanderVak) [reply olexandr.vakulinskiy/PR1/practice.rb]: Screenshot from 2020-12-09 23-39-48
I've been trying to fix this regex, here's the best version, but it matches some whitespaces, what can I do?

Comment (slenderock) [reply olexandr.vakulinskiy/PR1/practice.rb]: instead of parsing url u can parse response.content_type

Comment (AlexanderVak) [reply olexandr.vakulinskiy/PR1/practice.rb]: Oh, thanks, its very helpful

File: olexandr.vakulinskiy/PR1/practice.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    url_ext = response.headers['content-type'].split('/')[-1]
+
+    raise ArgumentError, "wrong url, status code #{response.status}" unless response.status == 200
+    raise TypeError, 'file is not an image' unless image?(url_ext)
+
+    File.open("#{file_name}.#{url_ext}", 'wb') { |fp| fp.write(response.body) }
+  end
+
+  def image?(url_ext)
+    %w[jpg jpeg png].include?(url_ext)
```
Comment (Flippylolz) [inline olexandr.vakulinskiy/PR1/practice.rb:15]: Actually there are much more image extensions, it's better to check the first part of content_type to be equal to image.

**Review by Flippylolz — APPROVED** [review]

**Review by slenderock — APPROVED** [review]

File: olexandr.vakulinskiy/PR1/practice.rb (hunk 3 above)
Comment (AlexanderVak) [reply olexandr.vakulinskiy/PR1/practice.rb:15]: Ok, I'll fix it, it will be fast

**Pull request merged**